lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Directive, HostListener, Output, EventEmitter, OnInit, OnDestroy } from '@angular/core';
+import { IdentityUserService } from '@alfresco/adf-core';
+import { ProcessCloudService } from '../services/process-cloud.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { ProcessInstanceCloud } from '../start-process/models/process-instance-cloud.model';
+
+@Directive({
+    selector: '[adf-cloud-cancel-process]'
+})
+export class CancelProcessDirective implements OnInit, OnDestroy {
+
+    /** Emitted when the process is cancelled. */
+    @Output()
+    success: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Emitted when the process cannot be cancelled. */
+    @Output()
+    error: EventEmitter<any> = new EventEmitter<any>();
+
+    @Output()
+    /** Emitted when the can cancel process status is checked */
+    canCancelProcess: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
not sure we need this event. The main goal of this directive is to change the beheviour of the html and disable/enable the button based on the canCancel.
I would suggest using something similar
`    this.renderer.setAttribute(this.el.nativeElement, 'disabled', 'true');`